services/requester/requester.go

```diff
@@ -373,16 +372,15 @@ func (e *EVM) EstimateGas(
 	if err != nil {
 		return 0, fmt.Errorf("failed to decode EVM result from gas estimation: %w", err)
 	}
+
 	if evmResult.ErrorCode != 0 {
+		if evmResult.ErrorCode == evmTypes.ExecutionErrCodeExecutionReverted {
+			return 0, errs.NewRevertError(evmResult.ReturnedData)
+		}
 		return 0, getErrorForCode(evmResult.ErrorCode)
```

Comment:
don't have this function now inside evm core?

Reply:
Oh, good point, I forgot that it was added :innocent:

Reply:
Updated in https://github.com/onflow/flow-evm-gateway/pull/303/commits/2192050e4b3b183ac4676a83a4ba91842afd10b4